Firestore/core/src/firebase/firestore/util/status.h

```diff
@@ -128,9 +134,40 @@ class ABSL_MUST_USE_RESULT Status {
     // NSError* to Status and back to NSError* losslessly.
     std::unique_ptr<PlatformError> platform_error;
   };
-  // OK status has a `nullptr` state_.  Otherwise, `state_` points to
-  // a `State` structure containing the error code and message(s)
-  std::unique_ptr<State> state_;
+
+  struct Deleter {
+    void operator()(const State* ptr) const {
+      if (ptr != MovedFromIndicator()) {
+        delete ptr;
+      }
+    }
+  };
+
+  // A `unique_ptr` with a custom deleter. If the pointer's value has been set
+  // to a special value (0x1) to indicate it is moved, invoking the custom
+  // deleter will be a no-op.
+  using StatePtr = std::unique_ptr<State, Deleter>;
+
+  static State* MovedFromIndicator() {
+    return reinterpret_cast<State*>(0x01);
+  }
+
+  static StatePtr MakeStatePtrWithMovedIndicator() {
+    return StatePtr(MovedFromIndicator());
+  }
+
+  template <typename... Args>
+  static StatePtr MakeStatePtr(Args... args) {
+    return StatePtr(new State(args...));
+  }
+
+  // OK status has a `nullptr` state_, and `IsMoved() == false`.  Otherwise,
+  // either `state_` points to a `State` structure containing the error code
+  // and message(s), or this instance has been moved.
+  StatePtr state_;
+
+  // Tags this instance as `moved_from`.
```

Comment:
Nit: `moved_from` looks like it's the name of a variable or a function, whereas I think it's actually just a concept. Replace with "moved-from" with no backticks?

Reply:
Done.

Reply:
Can you please replace the backticks with quotes?